src/firefly/js/ui/Menu.jsx

```diff
@@ -94,26 +106,48 @@ export function Menu() {
         }
     }, [selected,ready]);
 
-    if (!ready) return <div/>;
 
     const menuTabItems = menuItems
         ?.filter(({action,type}) => (action!=='app_data.helpLoad' && type!=='COMMAND'))
         ?.filter(itemVisible)
         ?.filter((item) => item.type!=='COMMAND');
 
+
+    useEffect(() => {
+        setTabCountRendered(menuTabItems.length);
+    }, [menuTabItems.length]);
+
+    if (!ready) return <div/>;
+
     const helpItem= menuItems?.find(({action,type}) => (action==='app_data.helpLoad' && type==='COMMAND'));
 
-    const bCntAdd= showBgMonitor?2:1;
-    const size= getButtonSize(menuTabItems.length+bCntAdd,windowWidth);
+    const setElement= (key,e) => {
+        if (!e) return;
+        const {tabWidthObj}= tabWidths;
+        tabWidthObj[key]= Math.trunc(e.getBoundingClientRect()?.width ?? 0);
+        Object.keys(tabWidthObj).forEach( (key) => {
+            if (Number(key)>=menuTabItems.length) tabWidthObj[key]= undefined;
+        });
+    };
 
-    return (
-        <Stack direction='row' justifyContent={'space-between'}>
-            <MenuTabBarJoyTabBased {...{menuTabItems,size,selected,bCntAdd,windowWidth,dropDown}}/>
+    const showHelp= Boolean(helpItem);
+    const size= getButtonSize(tabWidths.tabWidthObj,tbarElement.element,lastButtonSize.size,
+        showBgMonitor,showHelp,showUserInfo);
+    lastButtonSize.size= size;
 
-            <Stack direction='row' divider={<Divider orientation='vertical' sx={{mx:0.5, my: 1}}/>}>
+    return (
+        <Stack direction='row' justifyContent={'space-between'}
+               ref={(c) => {
+                   if (c) {
+                       setTabBarElement(c);
+                   }
+               }}>
+            <MenuTabBarJoyTabBased {...{menuTabItems,size,selected,dropDown, setElement}}/>
+
+            <Stack direction='row' divider={<Divider orientation='vertical' sx={{mr:0.5, mt: 1.25, mb:3/4}}/>}>
```

Comment:
Thanks for fixing this - top and bottom margin now is perfect.

[Not a big deal] I'd keep mx:0.5 except for 1st divider which needs mr: 0.5 only - because now hover background of icon button looks misaligned.

Reply:
Thanks for catching that.  I changed it because it did not look good when the two dividers got close together. I wanted them to appear to merge but it was offset and did not look right. I will work with it more.